ext/sockets/sockets.c

```diff
@@ -875,26 +936,9 @@ PHP_FUNCTION(socket_close)
 	zval		*arg1;
 	php_socket	*php_sock;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "r", &arg1) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O", &arg1, socket_ce) == FAILURE) {
 		RETURN_THROWS();
 	}
-
-	if ((php_sock = (php_socket *)zend_fetch_resource(Z_RES_P(arg1), le_socket_name, le_socket)) == NULL) {
-		RETURN_THROWS();
-	}
-
-	if (!Z_ISUNDEF(php_sock->zstream)) {
-		php_stream *stream = NULL;
-		php_stream_from_zval_no_verify(stream, &php_sock->zstream);
-		if (stream != NULL) {
-			/* close & destroy stream, incl. removing it from the rsrc list;
-			 * resource stored in php_sock->zstream will become invalid */
-			php_stream_free(stream,
-					PHP_STREAM_FREE_KEEP_RSRC | PHP_STREAM_FREE_CLOSE |
-					(stream->is_persistent?PHP_STREAM_FREE_CLOSE_PERSISTENT:0));
-		}
-	}
-	zend_list_close(Z_RES_P(arg1));
```

Comment:
I'm a bit concerned about this, as the close operation may have an actual effect if shutdown was not used beforehand.

Reply:
Maybe @bwoebi can advise.

I probably also wouldn't deprecate this function even if we do convert it to a no-op.

Reply:
Yes, close must invalidate the socket/stream, i.e. php_sock->zstream shall still be freed and UNDEF'd. So that socket_close() actually removes the fd from the process.